server/src/main/java/org/opensearch/search/aggregations/MultiBucketConsumerService.java

```diff
@@ -153,10 +157,24 @@ public void accept(int value) {
                     );
                 }
             }
-            // check parent circuit breaker every 1024 calls
-            callCount++;
-            if ((callCount & 0x3FF) == 0) {
-                breaker.addEstimateBytesAndMaybeBreak(0, "allocated_buckets");
+            callCount.increment();
+            // tripping the circuit breaker for other threads in case of concurrent search
+            // if the circuit breaker has tripped for one of the threads already, more info
+            // can be found on: https://github.com/opensearch-project/OpenSearch/issues/7785
+            if (circuitBreakerTripped) {
+                throw new CircuitBreakingException(
+                    "Circuit breaker has tripped for one of the other threads",
+                    CircuitBreaker.Durability.PERMANENT
+                );
+            }
+            // check parent circuit breaker every 1024 to (1024 + available processors) calls
+            if ((callCount.sum() & 0x3FF) <= Runtime.getRuntime().availableProcessors()) {
```

Comment:
let's cache `Runtime.getRuntime().availableProcessors()` in the constructor of `MultiBucketConsumer`. Also I think we should use `OpenSearchExecutors.allocatedProcessors()` instead as that will return the assigned processors to OpenSearch process

Reply:
`Runtime.getRuntime().availableProcessors()` could be constant even

@sohami  I think the usage `Runtime.getRuntime().availableProcessors()` is correct here, because this is what `LongAdder` is using under the hood

Reply:
I agree with @reta, @sohami are u also on the same page on using `Runtime.getRuntime().availableProcessors()` rather than `OpenSearchExecutors.allocatedProcessors()`?

Reply:
>  because this is what `LongAdder` is using under the hood, 

Thanks for sharing, was unaware of this.